### PR TITLE
Cont FCU if get payload attribute fails

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

Engine API semantic correctness

**What does this PR do? Why is it needed?**

In the event where get payload attribute fails, we should still call FCU to update the EL client on the latest head. This is the correct semantic and an improvement upon current behavior. EL client should be updated with the current head, and get payload attribute failing should not block this

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

Took a long time to write unit test

### CODE REVIEW

File: beacon-chain/blockchain/execution_engine_test.go [hunk 1]
```diff
@@ -30,6 +31,68 @@ import (
...
+		WithStateGen(stategen.New(beaconDB, fcs)),
+		WithForkChoiceStore(fcs),
+		WithProposerIdsCache(cache.NewProposerPayloadIDsCache()),
+	}
+	service, err := NewService(ctx, opts...)
+	st, _ := util.DeterministicGenesisState(t, 10)
+	service.head = &head{
+		state: st,
+	}
+	require.NoError(t, err)
```
Comment (saolyn) [inline beacon-chain/blockchain/execution_engine_test.go:55]: I think this error check should be on line 51 before we try to use `service`

Comment (terencechain) [reply beacon-chain/blockchain/execution_engine_test.go:55]: Nice catch!

**Review by saolyn — APPROVED** [review]
lgtm

File: beacon-chain/blockchain/execution_engine.go [hunk 2]
```diff
@@ -70,7 +70,6 @@ func (s *Service) notifyForkchoiceUpdate(ctx context.Context, arg *notifyForkcho
 	hasAttr, attr, proposerId, err := s.getPayloadAttribute(ctx, arg.headState, nextSlot)
 	if err != nil {
 		log.WithError(err).Error("Could not get head payload attribute")
-		return nil, nil
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/execution_engine.go:73]: So if there was an error, wouldn't `attr` be nil? What is the expected result of sending the EL a nil forkchoice?

Comment (prestonvanloon) [reply beacon-chain/blockchain/execution_engine.go:73]: Discussed offline and verified with spec. `null` / `nil` is an acceptable parameter for this call.

https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#engine_forkchoiceupdatedv1

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**